subiquity/server/controllers/identity.py

```diff
@@ -79,18 +84,36 @@ def load_autoinstall_data(self, data):
                 crypted_password=data["password"],
             )
             self.model.add_user(identity_data)
-
-    @with_context()
-    async def apply_autoinstall_config(self, context=None):
-        if self.model.user:
             return
+
+        # The identity section is required except if (any):
+        # 1. a user-data section is provided
         if "user-data" in self.app.autoinstall_config:
             return
-        if self.app.base_model.target is None:
-            return
+        # 2. we are installing not-Server (Desktop)
         if self.app.base_model.source.current.variant != "server":
             return
-        raise Exception("neither identity nor user-data provided")
+        # 2.1 (see definition of _install_source_is_desktop)
+        if self._install_source_is_desktop():
+            return
+        # 3. we are only refreshing the reset partition
+        # (The identity controller doesn't figure this out until the apply
+        # step, so we are going to cheat and inspect the situation here)
+        storage_config = self.app.autoinstall_config.get("storage")
+        if (
+            storage_config is not None
+            and storage_config.get("layout") is not None
+            and storage_config["layout"].get("reset-partition-only")
+        ):
+            return
+        # 4. identity section is interactive
+        if self.interactive():
+            return
+        raise AutoinstallError("neither identity nor user-data provided")
+
+    @with_context()
+    async def apply_autoinstall_config(self, context=None):
+        pass
```

Comment:
I think this matches the default implementation provided in `SubiquityController` and can be removed.

Reply:
Ah good point, thanks.